routes/browse/blog/searchBlogs.js

```diff
@@ -8,21 +8,20 @@ exports.main = async kwargs => {
 	let { source, search, country, type } = req.query || {}
 
 	return conn.task(t => {
-		if(searchText.length){
-			return t.any(searchBlogQuery(searchText?.trim(), page, country, type)).then(async (results) => {
-				// console.log('resultsresults ', results)
+			return t.any(searchBlogQuery(searchText, page, country, type)).then(async (results) => {
 				return {
 					searchResults : results,
 					page,
 					total_pages : results[0]?.total_pages || 0,
 					totalRecords : results[0]?.total_records || 0
 				}
 			  
-			}).catch(err => console.log(err))
-		}
-		else return {
-			searchResults : []
-		}
+			}).catch(err => {
+				console.log(err);
```

Comment:
is there a way to not rely on an exception for normal control flow? this will 1) fill the logs with errors that are not errors and 2) hide actual errors by returning a valid result

Reply:
we should prevent errors from happening in the first place